OrbitQt/ProfilingTargetDialog.cpp

```diff
@@ -0,0 +1,493 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ProfilingTargetDialog.h"
+
+#include <absl/flags/declare.h>
+#include <absl/flags/flag.h>
+#include <absl/strings/str_format.h>
+#include <absl/time/time.h>
+#include <grpc/impl/codegen/connectivity_state.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/security/credentials.h>
+#include <grpcpp/support/channel_arguments.h>
+
+#include <QAbstractButton>
+#include <QAbstractItemView>
+#include <QAbstractState>
+#include <QFileDialog>
+#include <QFrame>
+#include <QHeaderView>
+#include <QHistoryState>
+#include <QItemSelectionModel>
+#include <QLabel>
+#include <QLineEdit>
+#include <QModelIndexList>
+#include <QObject>
+#include <QPushButton>
+#include <QRadioButton>
+#include <QSortFilterProxyModel>
+#include <QTableView>
+#include <QTimer>
+#include <QVariant>
+#include <QWidget>
+#include <Qt>
+#include <memory>
+#include <optional>
+#include <type_traits>
+#include <utility>
+#include <variant>
+
+#include "ConnectToStadiaWidget.h"
+#include "Connections.h"
+#include "MainThreadExecutor.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitClientServices/ProcessManager.h"
+#include "OverlayWidget.h"
+#include "Path.h"
+#include "ProcessItemModel.h"
+#include "TargetConfiguration.h"
+#include "process.pb.h"
+#include "servicedeploymanager.h"
+
+ABSL_DECLARE_FLAG(bool, local);
+
+namespace {
+const int kProcessesRowHeight = 19;
+const int kLocalTryConnectTimeoutMs = 1000;
+}  // namespace
+
+namespace orbit_qt {
+
+using orbit_grpc_protos::ProcessInfo;
+
+ProfilingTargetDialog::ProfilingTargetDialog(
+    SshConnectionArtifacts* ssh_connection_artifacts, MainThreadExecutor* main_thread_executor,
+    std::optional<ConnectionConfiguration> connection_configuration, QWidget* parent)
+    : QDialog(parent),
+      ui_(std::make_unique<Ui::ProfilingTargetDialog>()),
+      current_target_(TargetEnum::kFile),
+      main_thread_executor_(main_thread_executor),
+      local_grpc_port_(ssh_connection_artifacts->GetGrpcPort().grpc_port),
+      s_stadia_(&state_machine_),
+      s_s_history_(&s_stadia_),
```

Comment:
Why do you choose this name convention? Don't you like simply s_stadia_history, and so on?

Reply:
I was following loosely how its done in the docs https://doc.qt.io/qt-5/statemachine-api.html
There, the states all have this `s` prefix and if there are sub states, then they have prefix `ss`. I revised this since then and now all names are fully spelled out